### ISSUE TITLE
SEC 381 - Open Liberty SSL should pick up JDK cacerts when using SSL defaults

### ISSUE THREAD

**Issue by atosak** [OpenLiberty/open-liberty#9016]
What commercial Liberty edition does this feature belong in (pick one)?      

Open Liberty

Should this feature be included in the Bluemix buildpack?      Guidance: See existing Bluemix buildpack features here: https://github.ibm.com/was-liberty/WS-CD-Open/tree/integration/dev/build.image/editions/bluemix       No  

​

What SalesForce Skill/Mission Team and legacy RETAIN queue will be responsible for supporting this feature?      Guidance: See existing skills/teams and queues here: https://ibm.biz/WASQueue               Select an existing one if you can, with the agreement of the tech lead and both managers involved.                  If creating a new SalesForce skill/team, contact the L3 manager in the link above to have it created before approval.                Any other questions, use the contacts in the URL above to kick off a discussion           SalesForce: WAS: L3 Core Security    

RETAIN:     WL3SEC  

​

This feature will introduce a new default behavior for newly created Open Liberty servers that are using the default SSL configuration (where we only create a Keystore but no Truststore -and the certs in the keystore are also used to establish trust).  

The effect of this will be for newly created servers to automatically trust the 90 or so certificates that the OpenJ9 JDK populates in the cacerts file by default (as of Java SE 11).  This provides a more usable SSL experience for applications wanting to do outbound secure HTTP calls to sites that are considered trusted OOTB by the JDK.

​
Link to WAD: https://ibm.ent.box.com/file/523275575856

**Comment by acdemyers** [comment]
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## List of Steps to complete or get approvals / sign-offs for Onboarding to the Liberty release (GM date)
 
Instructions:
 - Do the actions below and mark them complete in the checklist when they are done.
 - Make sure all feature readiness approvers put the appropriate tag on the epic to indicate their approval.
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE **Before Development Starts or 8 weeks before Onboarding** 
- [ ] POC Design / WAD Review Scheduled (David Chang) or N/A. AHA Feature marked.
- [ ] POC Design / WAD Reviewed (Feature Owner) or N/A. AHA Feature marked.
- [ ] Complete any To-Do's from the POC Review, and create a new TO-DO in the AHA Feature for the "Design Approval Request" and assign it to Ian Robinson. Leave the date blank. (Feature Owner)
- [ ] Design / WAD Approval (Ian Robinson) or N/A. AHA Feature marked.
- [ ] No Design / No WAD Approval (Arthur De Magalhaes - cloud / Alasdair Nottingham - server) or N/A. AHA Feature marked.
- [ ] AHA Feature has link to Github epic or issue. Github epic has link back to AHA Feature. (Feature Owner or Backlog Subtribe PM).
- [ ] SVT Requirements identified. (Epic owner / Feature owner with SVT focal point)
- [ ] ID Requirements identified. (Epic owner / Feature owner with ID focal point)
- [ ] Create a child task of the epic entitled "FAT Approval Test Summary".  Add and fill in the template as described here: https://github.ibm.com/was-liberty/WS-CD-Open/wiki/Feature-Review-(Feature-Test-Summary-Process)
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE **3 weeks before Onboarding** 
- [ ] Identify all open source libraries that are changing or are new. Work with Legal Release Services (Cass Tucker or Release PM) to get open source cleared and approved. Or N/A. (Epic Owner).   New or changed open source impacts license and Certificate of Originality.
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE ** 3 weeks before Onboarding** 
- [ ] All new or changed PII messages are checked into the integration branch, before the last translation shipment out. (Epic Owner)
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE **2 weeks before Onboarding** 
- [ ] Implementation complete. (Epic owner / Feature owner) 
- [ ] All function tests complete. Ready for FAT Approval. (Epic owner / Feature owner)
- [ ] Review all known issues for Stop Ship. (Epic owner / Feature owner / PM)
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## APPROVALS with TARGET COMPLETION DATE **2 to 1 week before Onboarding** 
Prereq: You must have the Design Approval or No Design Approval on the AHA Feature.
- [ ] **Accessibility** - (G Scott Johnston). Accessibility testing is complete or N/A. Approver adds label focalApproved:accessibility to the Epic in Github.
- [ ] **FAT Liberty SOE** - (Kevin Smith). SOE FATS are running successfully or N/A . Approver adds label focalApproved:fat to the Epic in Github.
- [ ] **Globalization** (Marika Joannidis - Liberty / Simy Cheeran - tWAS). Translation is complete or N/A. TVT - complete or N/A. Approver adds label focalApproved:globalization to the Epic in Github.
- [ ] **ID** - (Kareen Deen). Documentation work is complete or N/A . Approver adds label focalApproved:id to the Epic in Github.
- [ ] **Performance** - (Jared Anderson). Performance testing is complete with no high severity defects or N/A . Approver adds label focalApproved:performance to the Epic in Github.
- [ ] **Serviceability** - (Don Bourne). Serviceability has been addressed. 
- [ ] **STE** - (Swati Kasundra). STE chart deck is complete or N/A . Approver adds label focalApproved:ste to the Epic in Github.
- [ ] **SVT** - (Greg Ecock - Cloud, Brian Hanczaryk- APS). SVT is complete or N/A . Approver adds label focalApproved:svt to the Epic in Github.
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## TARGET COMPLETION DATE **1 week before Onboarding** 
- [ ] No Stop Ship issues for the feature. (Epic owner / Feature owner / Release PM)
- [ ] Ship Readiness Review and Release Notes completed (Epic owner / Feature owner / Release PM)
- [ ] Github Epic and Epic's issues are closed / complete. All PRs are committed to the master branch. (Epic owner / Feature owner / Backlog Subtribe PM)
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## NOT REQUIRED FOR A FEATURE
- [ ] **OL Guides** - (Yee-Kang Chang). Assessment for OL Guides is complete or N/A. 
- [ ] **WDT** - (Leonard Theivendra). WDT work complete or N/A.
 
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## Related Deliverables TARGET COMPLETION DATE **General Availability** 
- [ ] Blog article writeup (Epic owner / Feature owner / Laura Cowen)

**Comment by acdemyers** [comment]
Aha feature:  https://bigblue.aha.io/features/SEC-381

**Comment by acdemyers** [comment]
#9166

**Comment by acdemyers** [comment]
#9186

**Comment by skasund** [comment]
L2 has requested STE slides for this feature.  The STE template can be found at the link below: 

https://ibm.box.com/s/1an42g7zdgmaj84w7dft0indqfgi8ffm

Please upload the completed slides to the same BOX folder.  Thanks!

**Comment by skasund** [comment]
Please reach out to @andrea-pichler for any STE approvals between 10/19-10/22.  I'll be back on 10/23.  Thanks!

**Comment by acdemyers** [comment]
> L2 has requested STE slides for this feature. The STE template can be found at the link below:
> 
> https://ibm.box.com/s/1an42g7zdgmaj84w7dft0indqfgi8ffm
> 
> Please upload the completed slides to the same BOX folder. Thanks!

STE has been uploaded to https://ibm.ent.box.com/file/546322359382

**Comment by skasund** [comment]
Thanks for the STE slides

**Comment by donbourne** [comment]
**Serviceability Approval Comment** - Please answer the following questions for serviceability approval:

1.  WAD -- does the WAD identify the most likely problems customers will see and identify how the feature will enable them to diagnose and solve those problems without resorting to raising a PMR?  Have these issues been addressed in the implementation?

2.  Test and Demo -- As part of the serviceability process we're asking feature teams to test and analyze common problem paths for serviceability and demo those problem paths to someone not involved in the development of the feature (eg. L2, test team, or another development team).  
a) What problem paths were tested and demonstrated? 
b) Who did you demo to?
c) Do the people you demo'd to agree that the serviceability of the demonstrated problem scenarios is sufficient to avoid PMRs for any problems customers are likely to encounter, or that L2 should be able to quickly address those problems without need to engage L3?

3.  SVT -- SVT team is often the first team to try new features and often encounters problems setting up and using them.  Note that we're not expecting SVT to do full serviceability testing -- just to sign-off on the serviceability of the problem paths they encountered.
a) Who conducted SVT tests for this feature?
b) Do they agree that the serviceability of the problems they encountered is sufficient to avoid PMRs, or that L2 should be able to quickly address those problems without need to engage L3?  

4.  Which L2 / L3 queues will handle PMRs for this feature?  Ensure they are present in the contact reference file and in the queue contact summary, and that the respective L2/L3 teams know they are supporting it.  Ask Don Bourne if you need links or more info.

**Comment by acdemyers** [comment]
@donbourne   The template for serviceability has been completed above see my answers in bold.

**Comment by gscottj** [comment]
This feature does not add any new or update any existing user interfaces in a way that requires Accessibility verification testing. Accessibility approval granted.

**Comment by marikaj123** [comment]
Approving since all translated messages are in Master.

**Comment by chirp1** [comment]
Hi! ID can make updates to the WebSphere Liberty documentation as appropriate to keep it technically correct. For the OL docs, because there is a backlog that we are working down, depending on where the OL issue ends up on the backlog, we might not have the material documented by 190012. If we don't, add enough material to the this feature's blog post as needed, and so that users can see an example of the function.

**Comment by tevans78** [comment]
Can't be `target:ga` without design approval.

**Comment by marikaj123** [comment]
Alaine, Thank you for confirming the feature did make the GM release 190012. When you can please close the feature.  Thank you.

**Comment by acdemyers** [comment]
Closing...
